### ISSUE TITLE
`--diagnostics` says `"running-as-root": false` when running as root

### ISSUE THREAD

**Issue by noyannus** [bit-team/backintime#1967]
Searching for the reason why BiT run as root does not back up my `/root` I stumbled on this. 

Run as root, `backintime --diagnostics` says that it does *not* run as root.

```
localhost:~ # whoami
root								   # <<<=====
localhost:~ # backintime --version
backintime 1.5.2
localhost:~ # backintime --diagnostics
QSettings::value: Empty key passed
{main
    "backintime": {
        "name": "Back In Time",
        "version": "1.5.2",
        "running-as-root": false,		#  <<<=====
        "latest-config-version": 6,
        "local-config-file": "/root/.config/backintime/config",
        "local-config-file-found": true,
        "global-config-file": "/etc/backintime/config",
        "global-config-file-found": false,
        "started-from": "/usr/share/backintime/common",
        "user-callback": "/root/.config/backintime/user-callback",
        "keyring-supported": false
    },
...
```
A bug in `--diagnostics` or in BiT proper?

Seen on 
	Operating System: openSUSE Tumbleweed-Slowroll 20241103
	KDE Plasma Version: 6.2.3
	KDE Frameworks Version: 6.7.0
	Qt Version: 6.8.0
	Kernel Version: 6.11.8-1-default (64-bit)
	Graphics Platform: X11

BiT was installed the SUSE way, via Discover or Yast Software.

**Comment by buhtz** [comment]
That is interesting. Usually BIT does check this via this line:

    pwd.getpwuid(os.getuid()) == 'root'

Can you please execute this on your shell (with the related python interpreter binary):

    # python -c "import pwd; import os; print(os.getuid()); print(pwd.getpwuid(os.getuid()))"

**Comment by noyannus** [comment]
Here you are:

```
localhost:~ # python -c "import pwd; import os; print(os.getuid()); print(pwd.getpwuid(os.getuid()))"
0
pwd.struct_passwd(pw_name='root', pw_passwd='x', pw_uid=0, pw_gid=0, pw_gecos='root', pw_dir='/root', pw_shell='/bin/bash')
```

Could it be related to #1968?

**Comment by buhtz** [comment]
> ```
> pwd.struct_passwd(pw_name='root', pw_passwd='x', pw_uid=0, pw_gid=0, pw_gecos='root', pw_dir='/root', pw_shell='/bin/bash')
> ```

That is a bug (introduced by me I think).
It should be:

    pwd.getpwuid(os.getuid()).pw_name == 'root'

Would you like to try a fix and pull request?

> Could it be related to #1968?

Definitely not.

**Comment by noyannus** [comment]
> Would you like to try a fix and pull request?

Autsch, erwischt! Mir fehlt dazu die Kompetenz (ich fummel mich immer nur in einzelne Problemchen rein, ohne systematische Kenntnisse. Den Jargon kann ich aber ganz gut imitieren :-) ). 

Außerdem scheint mir da im Repo was zu fehlen (ich nehme nicht an, dass `pwd.struct_passwd` dynamisch erzeugt wird, und bis zum 01.04. ist es noch weit):

![Screenshot_20241209_215041](https://github.com/user-attachments/assets/3bdc4d24-20b8-447c-bf94-d7a113b5d59b)

**Comment by buhtz** [comment]
Fixed by 5f7bcf5
